src/appleseed/renderer/modeling/object/meshobject.cpp

```diff
@@ -37,6 +37,7 @@
 
```

Comment:
I don't think there's duplicated handling of alpha maps in the PR.

Currently, alpha maps are used for two different things in appleseed. Removed unwanted parts of surfaces when intersecting rays with geometry and compositing transparency.

I think that it would be good to try to separate the two cases in the future and this PR goes in that direction. For object alpha maps I'm considering only the first case. Compositing transparency is best handled in materials and / or OSL shaders.

I extended the Object interface to support alpha maps. For some object types it makes sense to support them, for some others it's not clear if it's a good idea and for others alpha maps will not work.

Objects that can have alpha maps, need to implement the part of the interface that's related to alpha maps, like MeshObject.

I can see at some point using object alpha maps to clip unwanted parts of geometry, 
renaming the alpha maps inputs from materials to opacity maps or something like that and 
use them just for compositing transparency. 

So I only want to enable them for objects where it makes sense and not for all present and future object types by default.